src/test/java/com/rapid7/jul/SocketChannelReceiver.java

```diff
@@ -0,0 +1,93 @@
+package com.rapid7.jul;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.Collections;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Stub of logentries server
+ */
+public class SocketChannelReceiver {
+
+    private BlockingQueue<String> messagesReceived = new ArrayBlockingQueue<String>(10);
+    private final int port;
+    private ServerSocketChannel serverSocket;
+    private boolean isReady = false;
+
+    private SocketChannelReceiver(int port) {
+        this.port = port;
+    }
+
+    public static SocketChannelReceiver createAndStartReceiver(int port) throws Exception {
+        final SocketChannelReceiver receiver = new SocketChannelReceiver(port);
+        Thread receiverThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                receiver.start();
+            }
+        });
+        receiverThread.start();
```

Comment:
Thread is never stopped and you don't have a reference to it to perform cleanup either. I think you should ensure the thread is stopped when `close()` is called.

Reply:
@maquilanti-r7 One outstanding comment here